src/snapred/backend/data/GroceryService.py

```diff
@@ -190,52 +237,110 @@ def getCloneOfWorkspace(self, name: WorkspaceName, copy: WorkspaceName):
             ws = None
         return ws
 
-    def _updateNeutronCacheFromADS(self, runId: str, useLiteMode: bool):
+    ## FETCH METHODS
+    """
+    The fetch methods orchestrate finding data files, loading them into workspaces,
+    and preserving a cache to prevent re-loading the same data files.
+    """
+
+    def _fetchInstrumentDonor(self, runId: str, useLiteMode: bool) -> WorkspaceName:
+        self._updateInstrumentCacheFromADS(runId, useLiteMode)
+
+        key = self._key(runId, useLiteMode)
+        if self._loadedInstruments.get(key) is not None:
+            return self._loadedInstruments.get(key)
+
+        wsName = None
+        self._updateNeutronCacheFromADS(runId, useLiteMode)
+        if self._loadedRuns.get(key) is not None:
+            # If possible, use a cached neutron-data workspace as an instrument donor
+            wsName = self._createRawNeutronWorkspaceName(runId, useLiteMode)
```

Comment:
For "nexus-loaded" workspaces, the instrument location parameters are encoded in the nexus file (which is actually an hdf-5 format file, but with a special schema that it must adhere to).  If we were loading _all_ of our workspaces from nexus, there would be _no_ instrument donor issues at all (it would be completely transparent  :(  ).